autogpt_platform/backend/backend/integrations/webhooks/tenancy_test.py

```
@@ -0,0 +1,783 @@
+        telegram,
+        "Config",
+        lambda: SimpleNamespace(platform_base_url="https://example.com"),
+    )
+    monkeypatch.setattr(
+        telegram,
+        "get_redis_async",
+        AsyncMock(side_effect=RuntimeError("redis unavailable")),
+    )
+
+    with patch.object(manager, "_get_suitable_auto_webhook_locked", locked_setup):
+        with pytest.raises(WebhookSetupUnavailableError, match="safely lock"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)